backend/middleware/s3Uploader.js

```
@@ -0,0 +1,42 @@
+const AWS = require('aws-sdk');
+
+AWS.config.update({ region: 'us-east-1' });
+
+const s3 = new AWS.S3();
+
+const BUCKET_NAME = 'image-uploads-data';
+
+async function uploadBufferToS3(buffer, filename, mimeType = 'image/jpeg') {
+  const params = {
+    Bucket: BUCKET_NAME,
+    Key: `uploads/${filename}`,
+    Body: buffer,
+    ContentType: mimeType,
+  };
+
+  try {
+    await s3.upload(params).promise();
+    return true; // Return true on successful upload
+  } catch (error) {
+    console.error('Error uploading to S3:', error);
+    return false; // Return false on upload failure
+  }
```

instead of returning true or false, is there a way to propagate error